homeassistant/components/here_travel_time/config_flow.py

```diff
@@ -0,0 +1,363 @@
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.selector import EntitySelector, LocationSelector, selector
+
+from .const import (
+    ARRIVAL_TIME,
+    CONF_ARRIVAL,
+    CONF_DEPARTURE,
+    CONF_DESTINATION,
+    CONF_ORIGIN,
+    CONF_ROUTE_MODE,
+    CONF_TIME,
+    CONF_TIME_TYPE,
+    CONF_TRAFFIC_MODE,
+    DEFAULT_NAME,
+    DEPARTURE_TIME,
+    DOMAIN,
+    ROUTE_MODE_FASTEST,
+    ROUTE_MODES,
+    TIME_TYPES,
+    TRAFFIC_MODE_DISABLED,
+    TRAFFIC_MODE_ENABLED,
+    TRAFFIC_MODES,
+    TRAVEL_MODE_CAR,
+    TRAVEL_MODE_PUBLIC_TIME_TABLE,
+    TRAVEL_MODES,
+    UNITS,
+)
+from .sensor import (
+    CONF_DESTINATION_ENTITY_ID,
+    CONF_DESTINATION_LATITUDE,
+    CONF_DESTINATION_LONGITUDE,
+    CONF_ORIGIN_ENTITY_ID,
+    CONF_ORIGIN_LATITUDE,
+    CONF_ORIGIN_LONGITUDE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def is_dupe_import(
+    entry: config_entries.ConfigEntry,
+    user_input: dict[str, Any],
+    options: dict[str, Any],
+) -> bool:
+    """Return whether imported config already exists."""
+    # Check the main data keys
+    if any(
+        user_input[key] != entry.data[key]
+        for key in (CONF_API_KEY, CONF_MODE, CONF_NAME)
+    ):
+        return False
+
+    # Check origin
+    for key in (
+        CONF_DESTINATION,
+        CONF_ORIGIN,
+        CONF_DESTINATION_ENTITY_ID,
+        CONF_ORIGIN_ENTITY_ID,
+    ):
+        if user_input.get(key) != entry.data.get(key):
+            return False
+
+    # We have to check for options that don't have defaults
+    for key in (
+        CONF_TRAFFIC_MODE,
+        CONF_UNIT_SYSTEM,
+        CONF_ROUTE_MODE,
+        CONF_TIME_TYPE,
+        CONF_TIME,
+    ):
+        if options.get(key) != entry.options.get(key):
+            return False
+
+    return True
+
+
+def validate_api_key(api_key: str) -> None:
+    """Validate the user input allows us to connect."""
+    known_working_origin = [38.9, -77.04833]
+    known_working_destination = [39.0, -77.1]
+    RoutingApi(api_key).public_transport_timetable(
+        known_working_origin,
+        known_working_destination,
+        True,
+        [
+            RouteMode[ROUTE_MODE_FASTEST],
+            RouteMode[TRAVEL_MODE_CAR],
+            RouteMode[TRAFFIC_MODE_ENABLED],
+        ],
+        arrival=None,
+        departure="now",
+    )
+
+
+def get_user_step_schema(data: dict[str, Any]) -> vol.Schema:
+    """Get a populated schema or default."""
+    name = DEFAULT_NAME if data.get(CONF_NAME) is None else data.get(CONF_NAME)
+    mode = TRAVEL_MODE_CAR if data.get(CONF_MODE) is None else data.get(CONF_MODE)
+
+    return vol.Schema(
+        {
+            vol.Optional(CONF_NAME, default=name): cv.string,
+            vol.Required(CONF_API_KEY, default=data.get(CONF_API_KEY)): cv.string,
+            vol.Optional(CONF_MODE, default=mode): vol.In(TRAVEL_MODES),
+        }
+    )
```

Comment:
I believe this is equivalent to this:

```suggestion
    return vol.Schema(
        {
            vol.Optional(CONF_NAME, default=data.get(CONF_NAME, DEFAULT_NAME)): cv.string,
            vol.Required(CONF_API_KEY, default=data.get(CONF_API_KEY)): cv.string,
            vol.Optional(CONF_MODE, default=data.get(CONF_MODE, TRAVEL_MODE_CAR)): vol.In(TRAVEL_MODES),
        }
    )
```